src/site/content/en/blog/web-codecs/index.md

```diff
@@ -0,0 +1,361 @@
+| 5. Launch                                    | Not started                  |
+
+</div>
+
+## Video processing workflow
+
+Frames are the centerpiece in video processing, thus in WebCodecs most classes
+either consume or produce frames. Video encoders convert frames into encoded
+chunks. Video decoders do the opposite. Track readers turn video tracks into a
+sequence of frames. By design all these transformations happen asynchronously.
+WebCodecs API tries to keep the web responsive by keeping the heavy lifting of
+video processing off the main thread.
+Currently in WebCodecs the only way to show a frame on the page is to convert it
+into an
+[`ImageBitmap`](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap)
+and either draw the bitmap on a canvas or convert it into a `WebGL` texture.
+
+
+
+![image](insert_image_url_here)
+
+## WebCodecs in action
+
+### Encoding
+
+It all starts with a `VideoFrame`.  There are two ways to convert existing
+pictures into `VideoFrame` objects.
+
+The first is to create a frame directly from an
+[ImageBitmap](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap).
+Just call the `VideoFrame()` constructor and give it a bitmap and a presentation
+timestamp.
+
+```js
+let cnv = document.createElement("canvas");
+// draw something on the canvas
+...
+let bitmap = await createImageBitmap(cnv);
+let frame_from_bitmap = new VideoFrame(bitmap, { timestamp: 0 });
+```
+
+The second is to use `VideoTrackReader` to set a function that will be called
+each time a new frame appears in a
+[`MediaStreamTrack`](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack).
+This is useful when you need to capture a video stream from a camera or the
+screen.
+
+```js
+let frames_from_stream = [];
+let stream = await navigator.mediaDevices.getUserMedia({ ... });
+let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
+vtr.start((frame) => {
+  frames_from_stream.push(frame);
+});
+```
+
+No matter where they are coming from, frames can be encoded into
+`EncodedVideoChunk` objects with a `VideoEncoder`.
+
+Before encoding `VideoEncoder` needs to be given two javascript objects:
+
++   Init dictionary with two functions for handling encoded chunks and
+    errors. These functions are developer-defined and can't be changed after
+    they're passed to the `VideoEncoder` constructor..
++   Encoder configuration object, which contains parameters for the output
+    video stream. You can change these parameters later  by calling `configure()`.
+
+```js
+const init = {
+  output: handleChunk,
+  error: (e) => {
+    console.log(e.message);
+  }
+};
+
+let config = {
+  codec: "vp8",
+  width: 640,
+  height: 480,
+  bitrate: 8_000_000,     // 8 Mbps
+  framerate: 30,
+};
+
+let encoder = new VideoEncoder(init);
+encoder.configure(config);
+```
+
+After the encoder has been set up,  it's ready to start accepting frames. When
+frames are coming from a media stream, `VideoTrackReader` will pump frames into
```

Comment:
```suggestion
frames are coming from a media stream, the `VideoTrackReader` callback will pump frames into
```

Reply:
What is he trying to communicate? `VideoTrackReader` is an interface, not a callback.

Reply:
Is he trying to refer to the callback passed to `VideoTrackReader.start()`? If so, it needs to be that explicit. What he suggested is too big of a leap for people don't know anything about WebCodecs (which is anyone reading this article).

Reply:
Yeah, that's exactly what he means here.  How about this  
```suggestion
frames are coming from a media stream, the callback given to `VideoTrackReader.start()` will pump frames into
```

Feel free to reword it in whatever way you see fit.